packages/web3-eth-contract/test/integration/local_account/contract_deploy.test.ts

```diff
@@ -0,0 +1,167 @@
+import { GreeterBytecode, GreeterAbi } from '../../shared_fixtures/build/Greeter';
+import { getSystemTestProvider, createLocalAccount, isWs } from '../../fixtures/system_test_utils';
+import { Contract } from '../../../src';
+import { sleep } from '../../shared_fixtures/utils';
+
+describe('contract', () => {
+	describe('deploy', () => {
+		let contract: Contract<typeof GreeterAbi>;
+		let sendOptions: Record<string, unknown>;
+		let deployOptions: Record<string, unknown>;
+		let localAccount: Web3Account;
+		let web3: Web3;
+
+		beforeAll(async () => {
+			web3 = new Web3(getSystemTestProvider());
+			contract = new web3.eth.Contract(GreeterAbi) as unknown as Contract<typeof GreeterAbi>;
+			deployOptions = {
+				data: GreeterBytecode,
+				arguments: ['My Greeting'],
+			};
+			localAccount = await createLocalAccount(web3);
+			sendOptions = {
+				from: localAccount.address,
+				gas: '1000000',
+			};
+		});
+
+		describe('local account', () => {
+			it.each(['0x1', '0x2'])('should deploy contract %p', async txType => {
+				const acc = await createLocalAccount(web3);
+				const deployedContract = await contract.deploy(deployOptions).send({
+					...sendOptions,
+					from: acc.address,
+					type: txType,
+				});
+				expect(deployedContract.options.address).toBeDefined();
+			});
+		});
+
+		it('deploy should fail with low baseFeeGas EIP1559', async () => {
+			await expect(
+				contract.deploy(deployOptions).send({
+					type: '0x2',
+					gas: '1000000',
+					maxFeePerGas: '0x1',
+					maxPriorityFeePerGas: '0x1',
+					from: localAccount.address,
+				}),
+			).rejects.toThrow(
+				"VM Exception while processing transaction: Transaction's maxFeePerGas",
+			);
+		});
+
+		it.each(['0x1', '0x2'])(
+			'should return estimated gas of contract method %p',
+			async txType => {
+				const contractDeployed = await contract.deploy(deployOptions).send(sendOptions);
+
+				const estimatedGas = await contractDeployed.methods
+					.setGreeting('Hello')
+					.estimateGas({
+						...sendOptions,
+						type: txType,
+					});
+				expect(Number(estimatedGas)).toBeGreaterThan(0);
+			},
+		);
+
+		it('should deploy the contract if data is provided at initiation', async () => {
+			const contractWithParams = new web3.eth.Contract(GreeterAbi, undefined, {
+				provider: web3.provider,
+				data: GreeterBytecode,
+				from: localAccount.address,
+				gas: '1000000',
+			}) as unknown as Contract<typeof GreeterAbi>;
+
+			const deployedContract = await contractWithParams
+				.deploy({ arguments: ['Hello World'] })
+				.send();
+
+			expect(deployedContract.options.address).toBeDefined();
+		});
+
+		it('should emit the "confirmation" event', async () => {
+			const confirmationHandler = jest.fn();
+
+			await contract
+				.deploy(deployOptions)
+				.send(sendOptions)
+				.on('confirmation', confirmationHandler);
+
+			// Wait for sometime to allow the transaction to be processed
+			await sleep(500);
```

Comment:
Except for the comments, LGTM.
Just a small suggestion. Instead of using `sleep`, can't we implement the login inside the handler and resolve inside the handler?

Reply:
we've already had the `deploy test`. I've just copied it and used the local wallet inside.